src/Microsoft.AspNetCore.Authentication/VirtualSchemeOptions.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    /// <summary>
+    /// Used to redirect authentication methods to another scheme
+    /// </summary>
+    public class VirtualSchemeOptions
+    {
+        public string DefaultTarget { get; set; }
+
+        public string AuthenticateTarget { get; set; }
+        public string ChallengeTarget { get; set; }
+        public string ForbidTarget { get; set; }
+        public string SignInTarget { get; set; }
+        public string SignOutTarget { get; set; }
+
+        /// <summary>
+        /// Used to select a default target scheme based on the request.
+        /// </summary>
+        public Func<HttpContext, string> DefaultTargetSelector { get; set; }
```

Comment:
does this need an additional parameter saying what action you're resolving so you can return different results for Challenge, Forbid, etc.? An enum maybe?

Reply:
You can always setup multiple virtual schemes and target that for each verb, i.e. "ChallengeScheme", "ForbidScheme" and use the selector like that, I want to avoid too much craziness if we can.

Reply:
Or I guess if we want to move away from funcs, we could make this a Type we resolve from DI and have a real interface and have it take an enum

Reply:
This still needs work. The context of the action you're trying to resolve matters.

Reply:
We can iterate on this on the next go round once we start working on the samples repo.  I think it'll be easier for us to play with having a basic default selector to start, and we can tweak it in the next PR, if we remove it entirely, you won't be able to do request based selection at all.